### PR TITLE
Feature/create base entities

### PR DESCRIPTION
**Pull request by osmolka**
Added Country, Company, Department and Employee entities. 
Added lombok.

### CODE REVIEW

File: src/main/java/com/itechart/hris/model/benefits/Department.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class Department {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private Long id;
+
+  private String name;
+
+  @OneToMany(mappedBy = "department", fetch = FetchType.LAZY)
+  private Set<Employee> employees;
```
Comment (avinahradau) [inline src/main/java/com/itechart/hris/model/benefits/Department.java]: More suitable cascade type here is PERSIST

File: src/main/java/com/itechart/hris/model/benefits/Country.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+public class Country {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private Long id;
+
+  private String name;
+
+  @OneToMany(mappedBy = "country", fetch = FetchType.LAZY)
+  private Set<Company> companies;
```
Comment (avinahradau) [inline src/main/java/com/itechart/hris/model/benefits/Country.java]: More suitable cascade type here is PERSIST

Comment (osmolka) [reply src/main/java/com/itechart/hris/model/benefits/Country.java]: Done

File: src/main/java/com/itechart/hris/model/benefits/Department.java (hunk 1 above)
Comment (osmolka) [reply src/main/java/com/itechart/hris/model/benefits/Department.java]: > More suitable cascade type here is PERSIST

Done

**Pull request merged**